src/VisualStudio/IntegrationTests/AbstractEditorTests.cs

```diff
@@ -0,0 +1,231 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+using Roslyn.Test.Utilities;
+using Roslyn.VisualStudio.Test.Utilities;
+using Roslyn.VisualStudio.Test.Utilities.Common;
+using Roslyn.VisualStudio.Test.Utilities.Input;
+using Roslyn.VisualStudio.Test.Utilities.OutOfProcess;
+using Xunit;
+
+namespace Roslyn.VisualStudio.IntegrationTests
+{
+    public abstract class AbstractEditorTests : IDisposable
+    {
+        private readonly VisualStudioInstanceContext _visualStudio;
+        private readonly VisualStudioWorkspace_OutOfProc _visualStudioWorkspace;
```

Comment:
there may be value in having the variables themselves state their in/out of procness.  I know i'd like it to be clear at the use site if i'm making an in-proc or out-proc call.